core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/ExcelInputFormat.java

```diff
@@ -0,0 +1,251 @@
+    return new ExcelRecordReader();
+  }
+
+  public static void setConfigurations(Job job, String filePattern, String sheetName, boolean reprocess,
+                                       int sheetNo, String columnList, boolean skipFirstRow, String terminateIfEmptyRow,
+                                       String rowLimit, String ifErrorRecord, String processedFiles) {
+
+    Configuration configuration = job.getConfiguration();
+    configuration.set(FILE_PATTERN, filePattern);
+    configuration.set(SHEET_NAME, sheetName);
+    configuration.setBoolean(RE_PROCESS, reprocess);
+    configuration.setInt(SHEET_NO, sheetNo);
+    configuration.set(COLUMN_LIST, columnList);
+    configuration.setBoolean(SKIP_FIRST_ROW, skipFirstRow);
+    configuration.set(TERMINATE_IF_EMPTY_ROW, terminateIfEmptyRow);
+
+    if (!Strings.isNullOrEmpty(rowLimit)) {
+      configuration.set(ROWS_LIMIT, rowLimit);
+    }
+
+    configuration.set(IF_ERROR_RECORD, ifErrorRecord);
+    configuration.set(PROCESSED_FILES, processedFiles);
+  }
+
+
+  /**
+   * Reads excel spread sheet, where the keys are the offset in the excel file and the text is the complete record.
+   */
+  public static class ExcelRecordReader extends RecordReader<LongWritable, Text> {
+
+    public static final String END = "END";
+    public static final String MID = "MID";
+    public static final String CELL_SEPERATION = "\t";
+    public static final String COLUMN_SEPERATION = "\r";
+
+    // Map key that represents the row index.
+    private LongWritable key;
+
+    // Map value that represents an excel row
+    private Text value;
+
+    // Represents a indvidual cell
+    private Cell cell;
+
+    // Specifies all the rows of an Excel spreadsheet - An iterator over all the rows.
+    private Iterator<Row> rows;
+
+    // InputStream handler for Excel files.
+    private FSDataInputStream fileIn;
+
+    // Path of input file.
+    private Path file;
+
+    // Specifies the row index.
+    private long rowIdx;
+
+    // Specifies last row num.
+    private int lastRowNum;
+
+    // Keeps row limits for each excel file(s)
+    private Map<String, Integer> fileRowLimitMap = new HashMap<>();
+
+
+    @Override
+    public void initialize(InputSplit genericSplit, TaskAttemptContext context) throws IOException,
+      InterruptedException {
+
+      FileSplit split = (FileSplit) genericSplit;
+      Configuration job = context.getConfiguration();
+      file = split.getPath();
+
+      FileSystem fs = file.getFileSystem(job);
+      fileIn = fs.open(split.getPath());
+
+      // Reads the excel file, selects the sheet to be read.
+
+      String sheetName = job.get(SHEET_NAME);
+      int sheetIndx = job.getInt(SHEET_NO, -1);
+
+      Sheet sheet = null; // sheet can be used as common for XSSF and HSSF workbook
+      try {
+        Workbook workbook = WorkbookFactory.create(fileIn);
+
+        if (sheetIndx < 0 && sheetName != null && !sheetName.isEmpty()) {
```

Comment:
Replaced input widget for accepting sheet name or sheet number. User can either specify sheet name or sheet index. By default, sheet number will be selected as input.